sql/cmd/cli/main.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Mishamba/EPAM_GOLANG_COURSE/sql/pkg/model"
+	"github.com/Mishamba/EPAM_GOLANG_COURSE/sql/pkg/repository/memory"
+)
+
+const (
+	CommandSave = iota + 1
+	CommandListAll
+	CommandGetByID
+	CommandGetByPhone
+	CommandGetByEmail
+	CommandSearchByName
+	CommandDelete
+)
+
+func main() {
+	repository := memory.NewContactsRepositoryInMemory()
+
+	for {
+		fmt.Print(menu)
+		var command int
+		if _, err := fmt.Scanf("%d", &command); err != nil {
+			log.Println(err)
+		}
+
+		switch command {
+		case CommandSave:
+			if err := Save(repository); err != nil {
+				log.Println(err)
+			}
+		case CommandListAll:
+			if err := ListAll(repository); err != nil {
+				log.Println(err)
+			}
+		case CommandGetByID:
+			if err := GetByID(repository); err != nil {
+				log.Println(err)
+			}
+		case CommandGetByPhone:
+			if err := GetByPhone(repository); err != nil {
+				log.Println(err)
+			}
+		case CommandGetByEmail:
+			if err := GetByEmail(repository); err != nil {
+				log.Println(err)
+			}
+		case CommandSearchByName:
+			if err := SearchByName(repository); err != nil {
+				log.Println(err)
+			}
+		case CommandDelete:
+			if err := Delete(repository); err != nil {
+				log.Println(err)
+			}
+		default:
+			log.Printf("command not foumd for value %d\n", command)
+		}
+
+		printSeparator()
+	}
+}
+
+func ListAll(rep model.ContactsRepository) error {
+	records, err := rep.ListAll()
+	if err != nil {
+		return fmt.Errorf("error in ListAll: %q", err.Error())
+	}
+
+	fmt.Println("ListAll:")
+	for _, r := range records {
+		fmt.Println(r)
+	}
+
+	return nil
+}
+
+func GetByID(rep model.ContactsRepository) error {
```

Comment:
If this func is called like `GetByID`, it should be kind of `GetByID(id uint)`

Reply:
so i need to rewrite all programm without using memory package?